src/DataCollection.Shared/Models/WrappedPopupFieldValue.cs

```diff
@@ -0,0 +1,111 @@
+﻿/*******************************************************************************
+  * Copyright 2019 Esri
+  *
+  *  Licensed under the Apache License, Version 2.0 (the "License");
+  *  you may not use this file except in compliance with the License.
+  *  You may obtain a copy of the License at
+  *
+  *  http://www.apache.org/licenses/LICENSE-2.0
+  *
+  *   Unless required by applicable law or agreed to in writing, software
+  *   distributed under the License is distributed on an "AS IS" BASIS,
+  *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *   See the License for the specific language governing permissions and
+  *   limitations under the License.
+******************************************************************************/
+
+using Esri.ArcGISRuntime.Mapping.Popups;
+using System;
+using System.ComponentModel;
+
+namespace Esri.ArcGISRuntime.ExampleApps.DataCollection.Shared.Models
+{
+    /// <summary>
+    /// Class wraps PopupFieldValue to enable proper data binding support with localized formatted date values.
+    /// Needed because FormattedValue formats dates in UTC, which causes awkward behavior when editing, since users expect local time zone.
+    /// A converter would work, but then you'd need to bind to the PopupField itself, which breaks property change notification for the FormattedValue property.
+    /// </summary>
+    public class WrappedPopupFieldValue : INotifyPropertyChanged
+    {
+        private PopupFieldValue _wrappedValue;
+
+        public WrappedPopupFieldValue(PopupFieldValue baseValue)
+        {
+            _wrappedValue = baseValue;
+            _wrappedValue.PropertyChanged += wrappedValue_propertyChanged;
+        }
+
+        private void wrappedValue_propertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            PropertyChanged?.Invoke(this, e);
+
+            // Also raise property change events for Formatted Value, because it may have changed.
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(ReformattedValue)));
+        }
+
+        /// <summary>
+        /// Converts dates before formatting the way <see cref="FormattedValue"/> would.
+        /// If you want times in UTC, bind to <see cref="FormattedValue"/> directly.
+        /// </summary>
+        public string ReformattedValue
+        {
+            get
+            {
+                if (Value is DateTimeOffset dateValue)
+                {
+                    var localDateTime = dateValue.ToLocalTime().DateTime;
+
+                    if (Field.Format?.DateFormat != null)
```

Comment:
You can omit the null check and switch on `Field.Format?.DateFormat`. A switch can handle null -- it will go to either `case null`, to `default`, or fall through to the end.